browser/ipfs/ipfs_host_resolver.cc

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+#include <vector>
+
+#include "brave/browser/ipfs/ipfs_host_resolver.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "net/base/host_port_pair.h"
+#include "net/dns/public/dns_protocol.h"
+#include "services/network/network_context.h"
+
+namespace {
+
+net::DnsConfigOverrides GetActualDnsConfig() {
+  SecureDnsConfig secure_dns_config =
+      SystemNetworkContextManager::GetStubResolverConfigReader()
+          ->GetSecureDnsConfiguration(true);
+
+  net::DnsConfigOverrides current_config_overrides;
+  current_config_overrides.search = std::vector<std::string>();
+  current_config_overrides.attempts = 1;
+
+  if (secure_dns_config.mode() == net::SecureDnsMode::kSecure) {
+    if (!secure_dns_config.servers().empty()) {
+      current_config_overrides.dns_over_https_servers.emplace(
+          secure_dns_config.servers());
+    }
+    current_config_overrides.secure_dns_mode = net::SecureDnsMode::kSecure;
+  } else {
+    // A DNS error that occurred in automatic mode must have had an insecure
```

Comment:
@diracdeltas this is from https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/net/dns_probe_service_factory.cc;l=244
probably same logic should work for our case as well. what do you think?

Reply:
i don't think we should do this because the chromium code only runs when there's already been a DNS error. the definition of `kOff` says it will try lookup without DoH as a fallback if there's an error: https://source.chromium.org/chromium/chromium/src/+/master:net/dns/public/secure_dns_mode.h;l=19;drc=04559eae8e59a730ab668d137cdc363ada8c6b24. the code here doesn't assume that there's already a DNS error.

i'm not familiar with this code, but it seems like `current_config_overrides.secure_dns_mode` in this case should just be `secure_dns_config.mode()` ?